File: Udemy/Algoritmos e Logica de Programacao/Java/ExerciciosJavaMatrizes/src/application/ProblemaSomaLinhas.java
package application;

import java.util.Locale;
import java.util.Scanner;

public class ProblemaSomaLinhas {

	public static void main(String[] args) {
		/*
		 * Problema "soma_linhas"
Fazer um programa para ler dois números inteiros M e N (máximo = 10). Em seguida, ler uma matriz
de M linhas e N colunas contendo números reais. Gerar um vetor de modo que cada elemento do vetor
seja a soma dos elementos da linha correspondente da matriz. Mostrar o vetor gerado.
		 * */
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
        System.out.print("Qual a quantidade de linhas da matriz? ");
        int m = sc.nextInt();
        System.out.print("Qual a quantidade de colunas da matriz? ");
        int n = sc.nextInt();
        double[][] mat = new double[m][n];
        double[] vet = new double[m];
        double soma = 0.0;
        for(int i = 0; i < m; i++)
        {
            System.out.println("Digite os elementos da "+(i+1)+"a. linha: ");
            for (int j=0; j < n; j++)
            {
                mat[i][j] = sc.nextDouble();
            }
        }

        System.out.println("VETOR GERADOR: ");

        for (int i = 0; i < m; i++)
        {
            soma = 0;
            for (int j = 0; j < n; j++)
            {
                soma = soma + mat[i][j];
            }
            vet[i] = soma;
        }

        for (int i = 0; i < m; i++)
        {
            System.out.println(vet[i]);
        }
		
		sc.close();

	}

}

File: Algoritmos e Logica de Programacao/C/Matrizes/ProblemaSomaLinhas.c
/*
Problema "soma_linhas"
Fazer um programa para ler dois números inteiros M e N (máximo = 10). Em seguida, ler uma matriz
de M linhas e N colunas contendo números reais. Gerar um vetor de modo que cada elemento do vetor
seja a soma dos elementos da linha correspondente da matriz. Mostrar o vetor gerado.
*/
#include<stdio.h>
#include<stdlib.h>
int main (){
	
	
	int m=0, n =0, i=0, j=0;
	double soma = 0.0;
	printf("Qual a quantidade de linhas da matriz? ");
	scanf("%d",&m);
	printf("Qual a quantidade de colunas da matriz? ");
	scanf("%d",&n);
	
	double vet[m][n], vetorG[m];
	
	for(i=0;i<m;i++){
		printf("Digite os elementos da %d a. linha: \n",i+1);
		for(j=0;j<n;j++){
			scanf("%lf",&vet[i][j]);
		}
	}
	
	for(i=0;i<m;i++){
		soma = 0;
		for(j=0;j<n;j++){
			soma = soma + vet[i][j];
		}
		vetorG[i] = soma;
	}
	
	 printf("VETOR GERADO:\n");
	
	for(i=0;i<m;i++){
		printf("%.1lf\n",vetorG[i]);
	}
	

    return 0;
}
